### ISSUE TITLE
[ColorBuffer] Can't copy to position <0, 0>

### ISSUE THREAD

**Issue by Ptit-Biscuit** [openrndr/openrndr#392]
### Operating System

Windows

### OPENRNDR version

0.4.4

### Java version (if applicable)

18

### Describe the bug

Hi :)

I'm trying to use `copyTo` from `ColorBufferGL3` to copy another buffer to position <0, 0> but for some reason it ends up on the bottom left corner.
When going with the `shadow` approch to set position <0, 0> everything is fine.

Any idea why?

PS: parameter `sourceTarget` seems mandatory otherwise it tries to use the other signature.

### Steps to reproduce the bug

```kotlin
fun main() = application {
    configure {
        width = 400
        height = 400
    }

    program {
        val buffer = colorBuffer(width, height).apply { fill(ColorRGBa.BLACK) }
        val test = colorBuffer(100, 100).apply { fill(ColorRGBa.RED) }

        extend {
            test.copyTo(
                buffer,
                sourceRectangle = IntRectangle(0, 0, test.width, test.height), // <- can't omit useless parameter?
                targetRectangle = IntRectangle(0, 0, test.width, test.height),
            )

            buffer.shadow.let {
                it.download()
                it[0, 0] = ColorRGBa.CYAN
                it.upload()
            }

            drawer.image(buffer)
        }
    }
}`

**Comment by hamoid** [comment]
Interesting :-)

It happens when copying from (0, 0) -> (0, 0). But these other combinations work fine:

- (0, 1) -> (0, 0)
- (1, 0) -> (0, 0)
- (0, 0) -> (0, 1)
- (0, 0) -> (1, 0)

So as soon as any of the values is not 0, it goes to the top left corner.

**Comment by Ptit-Biscuit** [comment]
@hamoid Yep, I didn't specify it but offsetting source or target position works fine :).

As for why `useTextSubImage` set to `true` copies buffer to the wrong place maybe [the flip part](https://github.com/openrndr/openrndr/blob/master/openrndr-jvm/openrndr-gl3/src/jvmMain/kotlin/org/openrndr/internal/gl3/ColorBufferGL3.kt#L462C17-L462C22) is one of the reason?

How can I test that? 🤔

**Comment by hamoid** [comment]
Yes, it is flipping it in those functions but I do not know what is the reason behind it. I guess in some situation it is required. Maybe @edwinRNDR can tell?

How to avoid the vertical flip?
